apps/antalmanac/src/lib/helpers.ts

```diff
@@ -18,4 +21,20 @@ export async function clickToCopy(event: MouseEvent<HTMLElement>, sectionCode: s
     openSnackbar('success', 'WebsocSection code copied to clipboard');
 }
 
+export function quickSearchForClasses(title: string, term: string) {
+    const decompCourseInfo: string[] | undefined = title.match(/^(.*)\s(\S+)$/)?.slice(1);
+    const tabNum = useTabStore.getState().activeTab;
+
+    // tabNum == 1 locks this feature to only the Search page (even on mobile)
+    if (decompCourseInfo && tabNum == 1) {
+        const deptIdx: number = depts.findIndex((item) => item.deptValue === decompCourseInfo[0]);
+        useQuickSearchStore.getState().setValue({
+            term: term,
+            deptLabel: depts[deptIdx].deptLabel,
+            deptValue: decompCourseInfo[0],
+            courseNumber: decompCourseInfo[1],
```

Comment:
I'm not sure what you mean. Like I don't have to use the regex to separate the deptValue and courseNumber?

Reply:
Ah sorry, didn't intend my comment to be so cryptic.

To rephrase: your regex's result isn't incorrect, however the usage of regex to re-derive the Department Code and Course Number via the course title is, put less eloquently, "jank".

The type you're working with, `CourseEvent`, at some point had certain fields removed from it (dept code, course number).

What I'm suggesting is that you identify where we omitted those fields from CourseEvent, and then include them so that you can have access to the values via CourseEvent. Then, no regex and re-deriving is needed.

Reply:
Ohh ok gotcha 👌